addons/mail/static/src/components/dialog_manager/dialog_manager.js

```diff
@@ -44,4 +39,4 @@ Object.assign(DialogManager, {
     template: 'mail.DialogManager',
 });
 
-registerMessagingComponent(DialogManager);
+registerMessagingComponent(DialogManager, { messagingField: 'dialogManager' });
```

Comment:
I don't like this. I would prefer that any registered messaging component could access messaging with `this.messaging` (or just `messaging` in template).

Reply:
I suppose component code will be slightly more noisy due to prefixing messaging in many places, but I think it's better than this cumbersome way of defining components that refer to core singleton records... The plan is to merge all of them in a single record (e.g. messaging) and split concern in a better way (by feature), so I don't mind everything with the prefix `messaging`.

Reply:
Ok for me, it doesn't change much. Less code when defining the component, but more characters when accessing sub fields.